### PR TITLE
feat(parser): support opening .ifcZIP containers

### LINKED ISSUE #1494: parser: support opening .ifcZIP containers
## Problem

`.ifcZIP` (the buildingSMART IFC container format — a zip archive wrapping a
single `.ifc`/`.ifcxml` and optionally referenced resources) is not supported
anywhere in the stack today.

Verified via a full-stack grep before filing this:
- `detectFormat()` (`packages/ifcx/src/index.ts`) only recognizes GLB (magic
  bytes), IFC STEP (`ISO-10303-21` header), and IFCX — no zip signature
  (`PK\x03\x04`) detection at all.
- Every other `.ifczip` hit in the codebase (`packages/mcp/src/loader.ts`,
  `apps/viewer/src/components/mcp/playground-dispatcher.ts`,
  `apps/viewer/src/lib/analytics-scrub.ts`,
  `apps/viewer/src/components/viewer/compare/useBcfFromChange.ts`) is a
  filename-extension regex used for display/naming/path-scrubbing only — no
  actual decompression.
- The only zip-reading dependency in the repo is `jszip` (already a
  dependency of `packages/bcf` and `packages/export`), used by
  `packages/bcf/src/reader.ts` to parse `.bcfzip`. There's no equivalent
  reader for `.ifcZIP`.

Today, a user must manually unzip and feed the bare `.ifc`/`.ifcxml` inside —
the library silently doesn't handle the container at all.

## Proposed scope (minimal)

1. **Format detection**: extend `detectFormat()` to recognize the zip magic
   (`PK\x03\x04`) as a candidate `.ifcZIP` container (distinguish from other
   zip-based formats we already handle, like `.bcfzip`, if they could ever
   reach the same entry point).
2. **Unwrap**: on detection, unzip (mirror `packages/bcf/src/reader.ts`'s
   `jszip` usage — same library, already a workspace dependency) and locate
   the single `.ifc`/`.ifcxml` entry inside (per the IFC container spec, one
   model file per archive; reject/error clearly if zero or multiple candidates
   are found rather than silently picking one).
3. Feed the extracted bytes through the existing `parseAuto`/`IfcParser` path
   unchanged — no new parsing logic needed once unwrapped.
4. Referenced resources (textures, documents) inside the container are
   out of scope for a first pass — note them as a known limitation.

## Where this needs wiring

- `packages/parser` (or `packages/ifcx`'s `detectFormat`, wherever the format
  sniff lives) — the core unwrap step.
- `apps/viewer` — drag-and-drop / file-open should accept `.ifczip` directly
  (currently only strips the extension for naming, per the grep above).
- `packages/cli` / `packages/mcp` (`loader.ts`) — same unwrap on file load.
- `packages/server-*` if it has its own upload/load path — check for parity.

## Out of scope (follow-up)

- Referenced external resources inside the container (textures/docs).
- Writing/exporting `.ifcZIP` (only opening is requested here).

### PR DESCRIPTION
**Pull request by Blogbotana**
Closes #1494

## Summary
- New `@ifc-lite/parser` exports: `isZipBuffer(buffer)` (magic-byte check) and `unwrapIfcZip(buffer)` — returns the single `.ifc`/`.ifcxml` entry's bytes if `buffer` is a zip container, or `buffer` unchanged otherwise. Safe to call unconditionally (cheap no-op for every ordinary file). Throws on zero or multiple candidate model entries rather than guessing.
- Mirrors the `jszip` usage already established in `packages/bcf/src/reader.ts` for `.bcfzip` — same dependency, same pattern, now also a direct dependency of `@ifc-lite/parser`.
- Wired into: `parseAuto`, the CLI loader (`loadIfcFile`), the MCP loader (`loadIfcModel`), and the viewer's `useIfcLoader` (single/multi-model load path — this covers regular multi-file federation too, since each file goes through the same `loadFile` call).
- Viewer file-picker `accept` attributes and drag-and-drop extension checks (`MainToolbar`, `MobileToolbar`, `ViewportContainer`, `McpPlayground`) now include `.ifczip`.

## Out of scope (noted, not fixed)
- Referenced resources inside the container (textures, documents) — only the model file's bytes are extracted.
- `useIfcFederation.ts`'s IFCX-specific federation path (already IFCX-only, zipping an IFCX file is not the common case the buildingSMART container format targets).

## Test plan
- [x] New `packages/parser/src/ifczip.test.ts` (9 cases): non-zip passthrough, extraction, case-insensitivity, nested paths, ignoring non-model resources, zero-entry / multi-entry error cases. All green, plus the full existing `packages/parser` suite (210 tests) unaffected.
- [x] `tsc --noEmit` clean on `packages/parser`, `packages/cli`, `packages/mcp`, `apps/viewer`.
- [x] `packages/cli`/`packages/mcp` test suites unaffected (109/109, 49/50 — the one pre-existing failure is an unrelated locale-formatting test in `mcp/tools/util.test.ts`, not touched by this change).
- [x] Real end-to-end smoke test: zipped a real sample IFC (`Ifc4_SampleHouse.ifc`) into a `.ifczip`, ran `ifc-lite info sample.ifczip` via the built CLI — parsed correctly (47,309 entities, storeys/elements listed).

## Summary by CodeRabbit

- **New Features**
  - Added transparent support for opening `.ifcZIP`/`.ifczip` containers across the parser, CLI, MCP loader, and viewer.
  - File pickers and drag-and-drop now accept `.ifczip` (including updated browser/mobile upload filters).
- **Bug Fixes**
  - Enhanced ZIP unwrapping safety: validates ZIP structure, enforces exactly one embedded `.ifc`/`.ifcxml` model, and applies decompression/size limits.
- **Documentation**
  - Updated the parsing guide with `.ifcZIP` behavior for `parseAuto`, including that referenced resources aren’t extracted.
- **Tests**
  - Added tests for ZIP detection, unwrapping, single-entry rules, and size-limit/edge cases.

### CODE REVIEW

File: packages/parser/src/ifczip.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ * handles the ArrayBuffer-slice dance for a view that may not span its whole
+ * backing buffer, so callers just wrap the result in `Buffer.from(...)`.
+ */
+export async function unwrapIfcZipView(view: Uint8Array): Promise<ArrayBuffer> {
+  const arrayBuffer = view.buffer.slice(
+    view.byteOffset,
+    view.byteOffset + view.byteLength,
+  ) as ArrayBuffer;
+  return unwrapIfcZip(arrayBuffer);
+}
```
[the remark this answers is not in the document]
Comment (louistrue) [reply packages/parser/src/ifczip.ts:157]: @Blogbotana please check 👆

**Review by louistrue — DISMISSED** [review]

**Comment by louistrue** [comment]
Thx, @Blogbotana 👍
